src/option_parser.c

```diff
@@ -226,20 +226,21 @@ int string_parse_bool(const void *data, const char *s, void *ret)
         return success;
 }
 
-struct section *new_section(const char *name)
+struct section *get_or_create_section(const char *name)
 {
-        for (int i = 0; i < section_count; i++) {
-                if (STR_EQ(name, sections[i].name)) {
-                        DIE("Duplicated section in dunstrc detected.");
-                }
-        }
+        struct section *s = get_section(name);
+        if (!s) {
+                LOG_D("New section: [%s]", name);
```

Comment:
This was basically reimplementing `get_section()`, the only difference being `DIE()` if there already was one. The new part was split out from `add_entry()` now calls here every time to have a new section created or an existing one reused.